### PR TITLE
Add LRU to ES fqn

### PR DESCRIPTION
**Pull request by ayush-shah**
### Describe your changes :

Add LRU to es fqn lookup

#
### Type of change :

- [x] Bug fix
- [x] Improvement
- [x] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Frontend Preview (Screenshots) :
<p align="center">For frontend related change, please link screenshots of your changes preview! Optional for backend related changes.
</p>

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#

### CODE REVIEW

File: ingestion/src/metadata/ingestion/ometa/mixins/es_mixin.py [hunk 1]
```diff
@@ -38,6 +39,7 @@ class ESMixin(Generic[T]):
 
     fqdn_search = "/search/query?q=fullyQualifiedName:{fqn}&from={from_}&size={size}&index={index}"
 
+    @functools.lru_cache(maxsize=512)
```
Comment (harshach) [inline ingestion/src/metadata/ingestion/ometa/mixins/es_mixin.py:42]: remove the maxsize

File: ingestion/src/metadata/ingestion/ometa/mixins/es_mixin.py [hunk 2]
```diff
@@ -84,12 +90,10 @@ def es_search_from_fqn(
         )
 
         try:
-            entity_list = self._search_es_entity(
+            query_string, response = self._search_es_entity(
```
Comment (OnkarVO7) [inline ingestion/src/metadata/ingestion/ometa/mixins/es_mixin.py:93]: @ayush-shah any specific reason for returning the `query_string` from `_search_es_entity` method? I don't see it being used anywhere?

Comment (ayush-shah) [reply ingestion/src/metadata/ingestion/ometa/mixins/es_mixin.py:93]: this is for the LRU cache mechanism, it needs a tuple response having, one of the params and other being the response.

in this case, if it hits, the cache will automatically return the response

Comment (ayush-shah) [reply ingestion/src/metadata/ingestion/ometa/mixins/es_mixin.py:93]: let me make that change, LRU is running with that as well.

**Review by OnkarVO7 — APPROVED** [review]

**Pull request merged**